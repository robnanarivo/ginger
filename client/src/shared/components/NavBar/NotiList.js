import * as React from 'react';
import {
  Stack, Paper, Box, Container,
} from '@mui/material';
import NotiRow from './NotiRow';

const NotiList = (props) => {
  const { notiList } = props;
  if (notiList.length === 0) {
    return (
      <Stack
        direction="column-reverse"
        justifyContent="space-evenly"
        alignItems="stretch"
        spacing={1}
        margin={1}
      >
        <Paper
          variant="outlined"
          elevation={0}
          sx={{
            borderColor: '#0079D3',
          }}
        >
          <Box
            margin="10px"
            display="flex"
            justifyContent="space-between"
            alignItems="center"
          >
            <Container>
              You are all caught up!
            </Container>
          </Box>
        </Paper>
      </Stack>
    );
  }
  return (
    <Stack
      direction="column-reverse"
      justifyContent="space-evenly"
      alignItems="stretch"
      spacing={1}
      margin={1}
    >
      {
        notiList.map((noti) => (
          <NotiRow noti={noti} key={noti.notificationId} />
        ))
      }
    </Stack>
  );
};

export default NotiList;
