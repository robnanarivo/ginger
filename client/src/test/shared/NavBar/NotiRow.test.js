import React from 'react';
import renderer from 'react-test-renderer';
import NotiList from '../../../shared/components/NavBar/NotiList';
import 'jest-styled-components';

describe('NotiList unit test', () => {
  it('NotiList Snapshot test', () => {
    const notiList = [{
      notificationId: 123,
      type: 123,
    }];
    const tree = renderer
      .create(<NotiList notiList={notiList} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
