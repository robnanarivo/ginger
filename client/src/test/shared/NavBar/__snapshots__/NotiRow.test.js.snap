// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotiList unit test NotiList Snapshot test 1`] = `
<div
  className="css-y189bd-MuiStack-root"
>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1kcohtu-MuiPaper-root"
        >
          <div
            className="MuiBox-root css-13ypcga"
          >
            <div
              className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
            />
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
